packages/@ourworldindata/grapher/src/chart/GuidedChartUtils.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react"
+import { GrapherProgrammaticInterface, GrapherState } from "../core/Grapher.js"
+import { MultiDimDataPageConfig } from "@ourworldindata/utils"
+import { MultiDimDimensionChoices } from "@ourworldindata/types"
+
+export interface GuidedChartContextValue {
+    grapherStateRef: React.RefObject<GrapherState>
+    chartRef?: React.RefObject<HTMLDivElement>
+    onGuidedChartLinkClick?: (href: string) => void
+    registerMultiDim?: (registrationData: {
+        config: MultiDimDataPageConfig
+        updater: (newSettings: MultiDimDimensionChoices) => void
+    }) => void
+}
+
+export const GuidedChartContext =
+    React.createContext<GuidedChartContextValue | null>(null)
+
+/**
+ * If called within a `GuidedChartContext`, sets the context's `grapherStateRef`
+ * to a new `GrapherState` instance initialized with the provided config.
+ * If no context is available, returns a local ref initialized with the config.
+ * This is so the `GrapherState` can be controlled from a GuidedChart,
+ * but also allows for local usage when not
+ */
+export function useMaybeGlobalGrapherStateRef(
```

Comment:
This is used in site/MultiDim.tsx and Grapher.tsx

Instead of calls to useRef, they now call this, which will instantiate and attach the GrapherState to the GuidedChart ref instead of their local one, if the GuidedChartContext exists.

Otherwise, it's same as before.